### ISSUE TITLE
Issue when generating mock files for interfaces

### ISSUE THREAD

**Issue by jabeattie** [apollographql/apollo-ios#2608]
## Bug report

I'd like to preface this that the new mock generation workflow is generally a fantastic addition and I'm already convinced its a far better solution than manually maintaining JSON files etc.

However I've noticed while attempting to migrate over to version 1.0.x that generated mocks for types that have properties that use an interface don't seem to be generated correctly.

## Versions

Please fill in the versions you're currently using: 

- `apollo-ios` SDK version: 1.0.2
- Xcode version:  14.0.1
- Swift version: 5.7
- Package manager: SPM

## Steps to reproduce

We have a GraphQL interface:
```
interface Offer {
  ... some properties
}
```
And a type that has a property that is this interface:
```
type OfferMeta {
  offer: Offer
}
```

When generating mocks for this schema, we get the the below for the interface and the `OfferMeta` type
In MockObject+Interfaces.swift
```
public extension MockObject {
  typealias Offer = Interface
}
```
In OfferMeta+Mock.swift
```
public class OfferMeta: MockObject {
  public static let objectType: Object = Menus.Objects.OfferMeta
  public static let _mockFields = MockFields()
  public typealias MockValueCollectionType = Array<Mock<OfferMeta>>

  public struct MockFields {
    @Field<Offer>("offer") public var offer
  }
}

public extension Mock where O == OfferMeta {
  convenience init(
    offer: AnyMock? = nil
  ) {
    self.init()
    self.offer = offer
  }
}
```

This fails to compile at the `self.offer = offer` line with this error:
```
Subscript 'subscript(dynamicMember:)' requires that 'Offer' conform to 'MockFieldValue'
```

However if I manually edit the MockFields object in the `OfferMeta` class to be this:
```
@Field<MockObject.Offer>("offer") public var offer
```

Then our mocks compile correctly and work correctly when passing in a mock type that conforms to that interface in the schema.

Should the mock compilation correctly prefix that Offer type with the MockObject parent class?
Or is there perhaps something elsewhere wrong with our setup that is causing this.

Thanks in advance

**Comment by calvincestari** [comment]
Hi @jabeattie 👋🏻 - thanks for raising this issue.

I've tried to replicate this in a sample project and believe I've got the same output as you mention - [issue2608.zip](https://github.com/apollographql/apollo-ios/files/9931909/issue2608.zip). It builds as expected though; using `1.0.2` or `1.0.3` makes no difference. Can you take a look at the sample project and let me know what is missing to reproduce the compiler error? Thanks.

**Comment by jabeattie** [comment]
Thanks for the response @calvincestari

When comparing the project supplied with our overall source code I've realised the issue was caused by a type conflict as there was an existing `Offer` struct being imported from the module where the GraphQL schema and generated code lives.

Turns out Xcode was less than helpful and just linked the `Offer` in the error to the one in the mocks, but never suggested there was another type.

I'll close this as I can rectify this in our code.
